index.js

```diff
@@ -14,26 +30,36 @@ module.exports = (template, data) => {
 	const doubleBraceRegex = /{{(\d+|[a-z$_][a-z\d$_]*?(?:\.[a-z\d$_]*?)*?)}}/gi;
 
 	if (doubleBraceRegex.test(template)) {
-		template = template.replace(doubleBraceRegex, (_, key) => {
-			let result = data;
+		template = template.replace(doubleBraceRegex, (placeholder, key) => {
+			const value = key.split('.').reduce((value, key) => value ? value[key] : undefined, data);
+			const transformedValue = transform({value, key});
+			if (transformedValue === undefined) {
+				if (ignoreMissing) {
+					return placeholder;
+				}
 
-			for (const property of key.split('.')) {
-				result = result ? result[property] : '';
+				throw new MissingValueError(key);
 			}
 
-			return htmlEscape(String(result));
+			return htmlEscape(String(transformedValue));
 		});
```

Comment:
The only difference with this block is that it escapes. I think the logic could be unified.

Reply:
I think this should be done by another PR for refactor, because it's quite different task than "add options".

Reply:
These functions being duplicates was fine before when the logic was simple (just a for-loop). You're adding more complexity to it now and I think it should be moved out.